gobblin-metastore/src/main/java/org/apache/gobblin/metastore/MysqlStateStore.java

```diff
@@ -73,7 +74,7 @@
  *
  * @param <T> state object type
  **/
-public class MysqlStateStore<T extends State> implements StateStore<T> {
+public class MysqlStateStore<T extends State> implements StateStore<T>, DatasetStateStore<T> {
```

Comment:
Instead of changing MySqlStateStore (and indeed, other state store implementations), can we have MySqlJobStatusStateStore implement DatasetStateStore interface? Won't that change be more localized?